.github/workflows/ci-tests.yml

```
@@ -0,0 +1,148 @@
+# This workflow builds a Zebra Docker image and runs integration and unit tests
+# on the Zebra codebase. It is designed to add the different test workflows
+name: Run tests
+
+# Ensures that only one workflow task will run at a time. Previous builds, if
+# already in process, will get cancelled. Only the latest commit will be allowed
+# to run, cancelling any workflows in between
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+on:
+  schedule:
+    # Run this job every Friday at mid-day UTC
+    # This is limited to the Zebra and lightwalletd Full Sync jobs
+    # TODO: we should move this behavior to a separate workflow
+    - cron: "0 12 * * 5"
+
+  workflow_dispatch:
+    inputs:
+      network:
+        default: "Mainnet"
+        description: "Network to deploy: Mainnet or Testnet"
+        required: true
+      regenerate-disks:
+        type: boolean
+        default: false
+        description: "Just run a Zebra checkpoint sync and update checkpoint disks"
+        required: true
+      run-full-sync:
+        type: boolean
+        default: false
+        description: "Just run a Zebra full sync on `network`, and update tip disks"
+        required: true
+      run-lwd-sync:
+        type: boolean
+        default: false
+        description: "Just run a lightwalletd full sync and update tip disks"
+        required: true
+      force_save_to_disk:
+        required: false
+        type: boolean
+        default: false
+        description: "Force tests to always create a cached state disk, if they already create disks"
+      no_cache:
+        description: "Disable the Docker cache for this build"
+        required: false
+        type: boolean
+        default: false
+
+  pull_request:
+    # Skip PRs where Rust code and dependencies aren't modified.
+    paths:
+      # code and tests
+      - "**/*.rs"
+      # hard-coded checkpoints and proptest regressions
+      - "**/*.txt"
+      # test data snapshots
+      - "**/*.snap"
+      # dependencies
+      - "**/Cargo.toml"
+      - "**/Cargo.lock"
+      # configuration files
+      - ".cargo/config.toml"
+      - "**/clippy.toml"
+      # workflow definitions
+      - "docker/**"
+      - ".dockerignore"
+      - ".github/workflows/ci-tests.yml"
```

Should we also add the following workflows?

```suggestion
      - ".github/workflows/ci-tests.yml"
      - ".github/workflows/sub-ci-unit-tests-docker.yml"
      - ".github/workflows/sub-ci-integration-tests-gcp.yml"
```